### PR TITLE
Apache Arrow support

### PR DESCRIPTION
**Pull request by JesseMckinzie**
- Add support for writing to Apache Arrow IPC and Parquet files from the CLI and Python API
- Add documentation for writing to Arrow
- Add unit tests for writing to Arrow

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -273,6 +294,64 @@ if(USEGPU)
...
+		# Store the path in a CMake variable
+		set(PYARROW_INCLUDE_PATH "${PATH_OUTPUT}" CACHE INTERNAL "Path to pyarrow include directory")
+		
+		# Set filename for finding pyarrow
+		set(python_filename "ci-utils/find_pyarrow.py")
+		set(python_filepath "${CMAKE_CURRENT_SOURCE_DIR}/${python_filename}")
+
+		# Execute the Python script to find pyarrow library path
+		execute_process(
+			#COMMAND ${Python_EXECUTABLE} -c "import pyarrow as pa; path=pa.get_library_dirs()[0] + '/libarrow_python.dylib'; print(path)"
```
Comment (sameeul) [inline CMakeLists.txt:330]: remove old/stale code

File: CMakeLists.txt [hunk 2]
```diff
@@ -273,6 +294,64 @@ if(USEGPU)
 	endif()
 endif()
 
+ 
+if(USEARROW)
+
+	#list(APPEND Nyxus_LIBRARIES ${ARROW_LIB})
```
Comment (sameeul) [inline CMakeLists.txt:300]: remove old/stale code

File: CMakeLists.txt [hunk 3]
```diff
@@ -273,6 +294,64 @@ if(USEGPU)
...
+
+	if(PARQUET_LINK_SHARED)
+		list(APPEND Nyxus_LIBRARIES parquet_shared)
+	else()
+		list(APPEND Nyxus_LIBRARIES parquet_static)
+	endif()
+
+	if (BUILD_LIB)
+		# Find Python
+		find_package(Python REQUIRED COMPONENTS Interpreter Development)
```
Comment (sameeul) [inline CMakeLists.txt:312]: can you check if we can get away w/o using this? Since we already find Python at the beginning.

File: CMakeLists.txt [hunk 4]
```diff
@@ -273,6 +294,64 @@ if(USEGPU)
...
+		set(python_filepath "${CMAKE_CURRENT_SOURCE_DIR}/${python_filename}")
+
+		# Execute the Python script to find pyarrow library path
+		execute_process(
+			#COMMAND ${Python_EXECUTABLE} -c "import pyarrow as pa; path=pa.get_library_dirs()[0] + '/libarrow_python.dylib'; print(path)"
+			COMMAND ${Python_EXECUTABLE} ${python_filepath}
+			OUTPUT_VARIABLE LIB_PATH_OUTPUT
+			OUTPUT_STRIP_TRAILING_WHITESPACE
+		)
+		#find_package(arrow_python_shared REQUIRED)
```
Comment (sameeul) [inline CMakeLists.txt:335]: remove old/stale code

File: src/nyx/environment.h [hunk 5]
```diff
@@ -81,6 +87,9 @@
 #define VERBOSITY_ROI_INFO 4
 #define VERBOSITY_DETAILED 8
 
+// Arrow ouput
+#define ARROW_OUPUT_TYPE "--arrowOutputType"
```
Comment (friskluft) [inline src/nyx/environment.h:91]: This define is declared in a wrong position in the file: it follows feature group nicknames. All the command line options are grouped together in lines 19-60.

Comment (friskluft) [reply src/nyx/environment.h:91]: What's critical: the help text issued by Environment::show_cmdline_help() doesn't mention the --arrowOutputType options. You need to edit show_cmdline_help()

**Review by friskluft — CHANGES_REQUESTED** [review]
I'd require editing Environment::show_cmdline_help() and grouping #define ARROW_OUPUT_TYPE "--arrowOutputType" together with the other CLI options. 
Speaking of functionality, we need to fix the PythonAPI to avoid creation of idle features data frame in Apache scenarios.

File: ci-utils/install_prereq_linux.sh [hunk 6]
```diff
@@ -171,3 +171,30 @@ cd build_man/
...
+        -DCMAKE_INSTALL_LIBDIR=lib \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DARROW_COMPUTE=ON \
+        -DARROW_CSV=ON \
+        -DARROW_DATASET=ON \
+        -DARROW_ACERO=ON \
+        .. 
+make -j4
+make install
+cd ../../
```
Comment (sameeul) [inline ci-utils/install_prereq_linux.sh:198]: Can you build and install pyarrow here also like Windows?

File: ci-utils/envs/conda_cpp.txt [hunk 7]
```diff
@@ -9,3 +9,5 @@ xsimd >=8,<9
 cmake
 dcmtk >=3.6.7
 fmjpeg2koj >=1.0.3
+arrow-cpp>=10.0.0
```
Comment (sameeul) [inline ci-utils/envs/conda_cpp.txt:12]: lint :-)

**Review by friskluft — APPROVED** [review]

**Pull request merged**